### PR TITLE
fix: set chain id in `useEthersWalletClient`

### PR DESCRIPTION
**Pull request by neekolas**
### <samp>🤖 Generated by Copilot at 993aea1</samp>

It seems that the `useEthersWalletClient` hook was throwing errors when changing chains. Because of the way the `signMessage` wrapper is implemented, when the wallet client is unavailable the code will default to an empty string. This then would throw an error in `xmtp-js`.

I added the current environment's chain to both the hook, and the WalletConnectConnector. I am not sure if you really need both (you definitely need the `WalletConnectConnector` update). With that, I am able to load my inbox using TrustWallet over WalletConnect 2.0 locally.

## Related issues

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### <samp>🤖 Generated by Copilot at 993aea1</samp>

* Import `CHAIN_ID` constant to use as network configuration parameter ([link](https://github.com/lensterxyz/lenster/pull/2976/files?diff=unified&w=0#diff-e3ca60b597c7227522c2b3d032d337a2f7c52933c1b1b09789ba31f389e36484R2))
* Add `chainId` option to `useWalletClient` hook and pass it to `WalletClient` constructor ([link](https://github.com/lensterxyz/lenster/pull/2976/files?diff=unified&w=0#diff-e3ca60b597c7227522c2b3d032d337a2f7c52933c1b1b09789ba31f389e36484L11-R12))

## Emoji

🌐🔗🪙

### CODE REVIEW

File: apps/web/src/components/utils/hooks/useEthersWalletClient.tsx [hunk 1]
```diff
@@ -8,7 +9,7 @@ const useEthersWalletClient = (): {
   };
   isLoading: boolean;
 } => {
-  const { data, isLoading } = useWalletClient();
+  const { data, isLoading } = useWalletClient({ chainId: CHAIN_ID });
```
Comment (bigint) [inline apps/web/src/components/utils/hooks/useEthersWalletClient.tsx:12]: So `useWalletClient` not picking the chain id automatically?

**Review by bigint — CHANGES_REQUESTED** [review]

File: apps/web/src/components/Common/Providers/index.tsx [hunk 2]
```diff
@@ -31,7 +31,8 @@ const { chains, publicClient } = configureChains(
 const connectors = [
   new InjectedConnector({ chains, options: { shimDisconnect: true } }),
   new WalletConnectConnector({
-    options: { projectId: WALLETCONNECT_PROJECT_ID }
+    options: { projectId: WALLETCONNECT_PROJECT_ID },
+    chains: [IS_MAINNET ? polygon : polygonMumbai, mainnet]
```
[the remark this answers is not in the document]
Comment (neekolas) [reply apps/web/src/components/Common/Providers/index.tsx:35]: I didn't try that. You're probably right that would work.

File: apps/web/src/components/utils/hooks/useEthersWalletClient.tsx (hunk 1 above)
Comment (neekolas) [reply apps/web/src/components/utils/hooks/useEthersWalletClient.tsx:12]: I'm not really sure why. I just know that after making these changes, I was able to load my inbox and stopped receiving signature errors.

**Comment by neekolas** [comment]
@bigint have you had a chance to test this. I'd like to merge today

**Pull request merged**